rust/src/daemon.rs

```diff
@@ -300,4 +347,38 @@ mod test {
             parse_override_source(input).unwrap_err();
         }
     }
+
+    #[test]
+    fn test_parse_revision() {
+        let ok_cases = [
+            (
+                "revision=078eea7e710630c0c8aec1071d3529fc307d8772f2ca3435fdc947a39c5a12bf",
+                (
+                    ParsedRevisionKind::Checksum,
+                    "078eea7e710630c0c8aec1071d3529fc307d8772f2ca3435fdc947a39c5a12bf",
+                ),
+            ),
+            (
+                "078eea7e710630c0c8aec1071d3529fc307d8772f2ca3435fdc947a39c5a12bf",
+                (
+                    ParsedRevisionKind::Checksum,
+                    "078eea7e710630c0c8aec1071d3529fc307d8772f2ca3435fdc947a39c5a12bf",
+                ),
+            ),
+            ("version=foo", (ParsedRevisionKind::Version, "foo")),
+            ("version=", (ParsedRevisionKind::Version, "")),
+            ("foo", (ParsedRevisionKind::Version, "foo")),
+            ("", (ParsedRevisionKind::Version, "")),
```

Comment:
Yeah, I don't think that was intentional.

Reply:
Ack, I added one more commit taking care of that (and the additional testcase too).

Reply:
I was curious about this so dug a bit more. AFAICT, nothing in ostree prevents creating commits with an empty version string. It's just another string metadata key and those support empty strings.

So... not sure. I think it's highly dubious for any content provider to do this, but since it's theoretically possible, it feels like on the client side we should still support it.

Maybe @cgwalters has a different opinion.

Reply:
Agreed, probably ostree should have enforced non-empty strings there.

(Also there's a larger problem of non-unique version strings being probably a big mistake on the builder side, and there can be only one `""`)

Reply:
To clarify I have no strong opinion on handling empty version strings.  I am fine errorring out.  I am fine continuing to try to support them.

I seriously doubt anyone is somehow relying on them though.

Reply:
Well ok, I dropped the second commit for now. There is value in just having a 1:1 port in here, and I can maybe go chasing this on the libostree writing side first.